### PR TITLE
MM-16935 - Capture docker logs

### PR DESCRIPTION
**Pull request by ali-farooq0**
### Summary
Adding a step to jenkins clean up to record docker logs to help investigate https://mattermost.atlassian.net/browse/MM-16935

### CODE REVIEW

**Comment by ali-farooq0** [comment]
Link to logs: https://build.mattermost.com/job/mp/job/mattermost-server-pr-new/job/PR-11735/6/artifact/docker_logs.zip/*view*

File: build/Jenkinsfile.pr [hunk 1]
```diff
@@ -293,9 +293,16 @@ pipeline {
             dir('src/github.com/mattermost/mattermost-server/build') {
                 ansiColor('xterm') {
                     sh """
+                        # capture docker logs
+                        docker ps -aq | xargs -I % sh -c 'docker ps -a --no-trunc -f id=% > %_docker_log; echo >> %_docker_log; docker logs --details -t % >> %_docker_log'
+                        docker ps -a --no-trunc > docker_ps
+                        docker stats -a --no-stream > docker_stats
+                        zip -qr docker_logs.zip *_docker_log docker_ps docker_stats
```
Comment (cpanato) [inline build/Jenkinsfile.pr]: this looks great and improves the debugging when needed.
however, as is this is collecting all the docker logs from the machine and you will need to filter out those that are for the build.
if possible can you get the logs only for running docker for the build?
maybe use  `docker-compose logs <<name of the service>>`

Comment (ali-farooq0) [reply build/Jenkinsfile.pr]: Good call. Updated.

**Comment by ali-farooq0** [comment]
```
// docker-compose_logs

Attaching to f5ce58e6-3677-4591-a328-d6edc1cee2d5-7_minio_1_3bdf7e06212d, f5ce58e6-3677-4591-a328-d6edc1cee2d5-7_elasticsearch_1_a20fea0996fe, f5ce58e6-3677-4591-a328-d6edc1cee2d5-7_postgres_1_fd25b5a14e62, f5ce58e6-3677-4591-a328-d6edc1cee2d5-7_redis_1_d797b0498321, f5ce58e6-3677-4591-a328-d6edc1cee2d5-7_inbucket_1_ea8a7d17b026, f5ce58e6-3677-4591-a328-d6edc1cee2d5-7_mysql_1_f03fe0e8da11, f5ce58e6-3677-4591-a328-d6edc1cee2d5-7_openldap_1_71d8425d7c51
elasticsearch_1_a20fea0996fe | 2019-07-31T11:16:26.273223037Z OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.
elasticsearch_1_a20fea0996fe | 2019-07-31T11:16:28.166516193Z [2019-07-31T11:16:28,163][WARN ][o.e.c.l.LogConfigurator  ] [unknown] Some logging configurations have %marker but don't have %node_name. We will automatically add %node_name to the pattern to ease the migration for users who customize log4j2.properties but will stop this behavior in 7.0. You should manually replace `%node_name` with `[%node_name]%marker ` in these locations:
elasticsearch_1_a20fea0996fe | 2019-07-31T11:16:28.166591321Z   /usr/share/elasticsearch/config/log4j2.properties
...
```

```
// docker_stats

CONTAINER           CPU %               MEM USAGE / LIMIT       MEM %               NET I/O             BLOCK I/O           PIDS
27df8359dcd4        0.01%               95.98 MiB / 29.44 GiB   0.32%               4.51 MB / 3.96 MB   0 B / 8.19 kB       6
4e2bfdb395f2        0.00%               754.6 MiB / 29.44 GiB   2.50%               138 kB / 348 kB     1.55 MB / 13.6 MB   9
ee51feeb6296        0.06%               823.1 MiB / 29.44 GiB   2.73%               288 MB / 665 MB     0 B / 36.9 kB       44
...
```

```
// docker_ps

CONTAINER ID                                                       IMAGE                                              COMMAND                                                             CREATED             STATUS                    PORTS                             NAMES
27df8359dcd44f06f3ddb533cba8630f5f7c8d2d3cd9f4a3c640a2447eba2307   postgres:9.4                                       "docker-entrypoint.sh postgres"                                     13 minutes ago      Up 13 minutes             5432/tcp                          02a81215-0c66-45a9-aaae-64c07106e33b-2_postgres_1_85d5d640cdde
4e2bfdb395f2374a9ecbc0e5e5105646a43917286eb19245fb07ffcd8cb7d86b   osixia/openldap:1.2.2                              "/container/tool/run"                                               13 minutes ago      Up 13 minutes             389/tcp, 636/tcp                  02a81215-0c66-45a9-aaae-64c07106e33b-2_openldap_1_66e4892142de
ee51feeb6296205f804095c81bee0175c7beeabc7372d0227d9dd5647dff780b   mysql:5.7                                          "docker-entrypoint.sh mysqld"                                       13 minutes ago      Up 13 minutes             3306/tcp, 33060/tcp               02a81215-0c66-45a9-aaae-64c07106e33b-2_mysql_1_d03d68de8288
a68d78e02f4af40910a6de5fefda1a1cbf51c5ee5682a77e08e778217b056f2c   mattermost/mattermost-elasticsearch-docker:6.5.1   "/usr/local/bin/docker-entrypoint.sh eswrapper"                     13 minutes ago      Up 13 minutes             9200/tcp, 9300/tcp                02a81215-0c66-45a9-aaae-64c07106e33b-2_elasticsearch_1_d6fa8082cbee
e0c8ff4628672d17714ffcbda2455d25cff5841a97186c109bb674c35407fa4e   minio/minio:RELEASE.2019-04-23T23-50-36Z           "/usr/bin/docker-entrypoint.sh server /data"                        13 minutes ago      Up 13 minutes (healthy)   9000/tcp                          02a81215-0c66-45a9-aaae-64c07106e33b-2_minio_1_a0cba7bc4a45
9ad631adde94f42541860501abb408119c6cfc47e545a0438126e2807ddc7d36   jhillyerd/inbucket:release-1.2.0                   "/con/context/start-inbucket.sh /con/configuration/inbucket.conf"   13 minutes ago      Up 13 minutes             10025/tcp, 10080/tcp, 10110/tcp   02a81215-0c66-45a9-aaae-64c07106e33b-2_inbucket_1_a711cbafe945
45fbf916af90e5147ff630c7f177af67785ad5e0eec44f3121445856c3dcba48   redis                                              "docker-entrypoint.sh redis-server"                                 13 minutes ago      Up 13 minutes             6379/tcp                          02a81215-0c66-45a9-aaae-64c07106e33b-2_redis_1_81c9da215c5e
66993bf600edd4347d3da17e8bafa23a176ff631a100c02004c493a76ca4ac66   minio/minio:RELEASE.2019-04-23T23-50-36Z           "/usr/bin/docker-entrypoint.sh server /data"                        16 minutes ago      Up 16 minutes (healthy)   9000/tcp                          f5ce58e6-3677-4591-a328-d6edc1cee2d5-7_minio_1_3bdf7e06212d
9021415ac4570a39c59c6f47a15a3b70849b507bd76cd856b945b329f644066b   mattermost/mattermost-elasticsearch-docker:6.5.1   "/usr/local/bin/docker-entrypoint.sh eswrapper"                     16 minutes ago      Up 16 minutes             9200/tcp, 9300/tcp                f5ce58e6-3677-4591-a328-d6edc1cee2d5-7_elasticsearch_1_a20fea0996fe
892dafe74da07305d99adcaeb418f65b0408fe7211366918cff887de8f88cd7c   postgres:9.4                                       "docker-entrypoint.sh postgres"                                     16 minutes ago      Up 16 minutes             5432/tcp                          f5ce58e6-3677-4591-a328-d6edc1cee2d5-7_postgres_1_fd25b5a14e62
05c000d6a6ea1618b712854724e34929b77345f0be173739503bdb39dd32a26d   redis                                              "docker-entrypoint.sh redis-server"                                 16 minutes ago      Up 16 minutes             6379/tcp                          f5ce58e6-3677-4591-a328-d6edc1cee2d5-7_redis_1_d797b0498321
671ec4ab6c20be517b20c9842258c75bd02cdbf103389c1af3820daaae846678   jhillyerd/inbucket:release-1.2.0                   "/con/context/start-inbucket.sh /con/configuration/inbucket.conf"   16 minutes ago      Up 16 minutes             10025/tcp, 10080/tcp, 10110/tcp   f5ce58e6-3677-4591-a328-d6edc1cee2d5-7_inbucket_1_ea8a7d17b026
fc561504afadf72c02e0de22ef93e61d04404b7672c6c76921b927988d2c1b2c   mysql:5.7                                          "docker-entrypoint.sh mysqld"                                       16 minutes ago      Up 16 minutes             3306/tcp, 33060/tcp               f5ce58e6-3677-4591-a328-d6edc1cee2d5-7_mysql_1_f03fe0e8da11
0178e9b2c368382bb68284767aae5f55b15c3d69a29fb77e65d70cabc21e3f70   osixia/openldap:1.2.2                              "/container/tool/run"                                               16 minutes ago      Up 16 minutes             389/tcp, 636/tcp                  f5ce58e6-3677-4591-a328-d6edc1cee2d5-7_openldap_1_71d8425d7c51
```

File: build/Jenkinsfile.pr (hunk 1 above)
Comment (cpanato) [reply build/Jenkinsfile.pr]: cool
that makes sense to you?

Comment (ali-farooq0) [reply build/Jenkinsfile.pr]: yes. already updated the pr to log containers for the current build run

File: build/Jenkinsfile.pr [hunk 2]
```diff
@@ -293,9 +293,16 @@ pipeline {
             dir('src/github.com/mattermost/mattermost-server/build') {
                 ansiColor('xterm') {
                     sh """
+                        # Capture docker logs
+                        docker-compose logs --tail="all" -t --no-color > docker-compose_logs
+                        docker ps -a --no-trunc > docker_ps
+                        docker stats -a --no-stream > docker_stats
+                        zip -qr docker_logs.zip docker-compose_logs docker_ps docker_stats
```
Comment (lieut-data) [inline build/Jenkinsfile.pr]: nit: can we `tar` this instead?

Comment (ali-farooq0) [reply build/Jenkinsfile.pr]: Done. https://build.mattermost.com/job/mp/job/mattermost-server-pr-new/job/docker_logs/10/artifact/docker_logs.tar.gz

**Pull request merged**